sql/analyzer/indexed_joins.go

```diff
@@ -378,6 +382,83 @@ func convertSemiToInnerJoin(a *Analyzer, m *Memo) error {
 	})
 }
 
+// convertAntiToLeftJoin adds left join alternatives for anti join
+func convertAntiToLeftJoin(a *Analyzer, m *Memo) error {
+	seen := make(map[GroupId]struct{})
+	return dfsExprGroup(m.root, m, seen, func(e relExpr) error {
+		anti, ok := e.(*antiJoin)
+		if !ok {
+			return nil
+		}
+
+		rightOutTables := m.tableProps.getTableNames(anti.right.relProps.OutputTables())
+		var projectExpressions []sql.Expression
+		onlyEquality := true
+		for _, f := range anti.filter {
+			transform.InspectExpr(f, func(e sql.Expression) bool {
+				switch e := e.(type) {
+				case *expression.GetField:
+					// getField expressions tell us which columns are used, so we can create the correct project
+					tableName := strings.ToLower(e.Table())
+					isRightOutTable := stringContains(rightOutTables, tableName)
+					if isRightOutTable {
+						projectExpressions = append(projectExpressions, e)
+					}
+				case *expression.Literal, *expression.BindVar,
+					*expression.And, *expression.Or, *expression.Equals, *expression.Arithmetic:
+					// these expressions are equality expressions
+				default:
+					onlyEquality = false
+				}
+				return !onlyEquality
+			})
+			if !onlyEquality {
+				return nil
+			}
+		}
+
+		// project is a new group
+		newRight := &project{
+			relBase:     &relBase{},
+			child:       anti.right,
+			projections: projectExpressions,
+		}
+		rightGrp := m.memoize(newRight)
+
+		// join is a new group
+		newJoin := &leftJoin{
+			joinBase: &joinBase{
+				relBase: &relBase{},
+				left:    anti.left,
+				right:   rightGrp,
+				op:      plan.JoinTypeLeftOuterHash,
+				filter:  anti.filter,
+			},
+		}
+		joinGrp := m.memoize(newJoin)
+
+		// drop null primary keys on right table
+		nullFilters := make([]sql.Expression, len(projectExpressions))
+		for i, e := range projectExpressions {
+			nullFilters[i] = expression.NewIsNull(e)
+		}
+		nullFilter := expression.JoinAnd(nullFilters...)
+		joinGrp.relProps.filter = nullFilter
```

Comment:
could we make this `projectExpressions`? need more testing to verify. alternatively, pk will be preferable to all columns for pk tables